addons/web/static/src/webclient/actions/action_service.js

```diff
@@ -971,10 +971,32 @@ function makeActionManager(env) {
             report_url: _getReportUrl(action, "html"),
             context: Object.assign({}, action.context),
         });
-        const clientActionOptions = Object.assign({}, options, {
-            props,
-        });
-        return doAction("report.client_action", clientActionOptions);
+
+        const clientAction = actionRegistry.get("report.client_action");
+        if (clientAction.prototype instanceof Component) {
+            if (action.target !== "new") {
+                await clearUncommittedChanges(env);
+                if (clientAction.target) {
+                    action.target = clientAction.target;
+                }
+            }
+            const controller = {
+                jsId: `controller_${++id}`,
+                Component: clientAction,
+                action,
+                ..._getActionInfo(action, props),
+            };
+            return _updateUI(controller, {
+                clearBreadcrumbs: options.clearBreadcrumbs,
+                stackPosition: options.stackPosition,
+                onClose: options.onClose,
+            });
+        } else {
+            const next = await clientAction(env, action);
+            if (next) {
+                return doAction(next, options);
+            }
+        }
```

Comment:
you don't need all this generic code. The ReportClientAction (which should be renamed, when converted to owl) is a simple Component (e.g. ReportComponent or something). No need to add it/retrieve it from the action registry: it can be directly imported. No need to handle the case where it is a function as we know it's a component. No need to handle the target "new" case, it's automatically in the default target.

Reply:
@aab-odoo I don't agree with `No need to add it/retrieve it from the action registry: it can be directly imported`. Currently, it's an `AbstractAction` transformed into a Component through `registerClientAction` (from `legacyActionRegistry.onAdd(registerClientAction);`). So until the refactoring of the report code, I need to pull it from the registry.